src/Iodev/Whois/Factory.php

```diff
@@ -37,7 +37,7 @@ public function createPunycode(): IPunycode
         if (function_exists("idn_to_utf8") && function_exists("idn_to_ascii")) {
```

Comment:
Try to delete this condition and `SymfonyPunycodeCovertor` class.
It should be enough for tests passing:
```
    public function createPunycode(): IPunycode
    {
        return new IntlPunycode();
    }
```

Reply:
Yes, it works without `idn_to*` checker.

Reply:
Nice!

Btw, `composer.lock` is not in git due to very different versions of php must be supported. I'm very not sure all versions from php 7.2 to 8.2  (+ intl/non-intl) will work correct if .lock file commited